cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/io/cdap/cdap/etl/batch/BatchPipelineSpecGenerator.java

```diff
@@ -67,7 +67,8 @@ public BatchPipelineSpec generateSpec(ETLBatchConfig config) throws ValidationEx
   }
 
   private StageSpec configureSqlEngine(ETLBatchConfig config) throws ValidationException {
-    if (config.getTransformationPushdown() == null || config.getTransformationPushdown().getPlugin() == null) {
+    if (!config.isPushdownEnabled() ||
+      (config.getTransformationPushdown() == null || config.getTransformationPushdown().getPlugin() == null)) {
```

Comment:
I don't think you need extra parenthesis here

Reply:
good point. I had written the logic differently originally, but let me fix this.